Source/WebCore/rendering/GridMasonryLayout.cpp

```diff
@@ -87,65 +87,77 @@ void GridMasonryLayout::resizeAndResetRunningPositions()
 
 void GridMasonryLayout::addItemsToFirstTrack(const HashMap<RenderBox*, GridArea>& firstTrackItems)
 {
-    for (auto& [item, gridArea] : firstTrackItems)
-        insertIntoGridAndLayoutItem(item, gridArea);
+    for (auto& [item, gridArea] : firstTrackItems) {
+        ASSERT(item);
+        if (!item)
+            continue;
+        insertIntoGridAndLayoutItem(*item, gridArea);
+    }
 }
 
 void GridMasonryLayout::placeItemsWithDefiniteGridAxisPosition(const Vector<RenderBox*>& itemsWithDefiniteGridAxisPosition)
```

Comment:
A vector of raw points should be an anti-pattern. I'd suggest a follow-up bug (future, for now) to work with WeakPtrs or CheckPtrs instead of bare pointers.

Reply:
std::reference_wrapper works for holding a container of references so that you don't have to null check on retrieval.

It's probably worth clearing the container at the end of masonryLayout so that it's super obvious that we're not holding these beyond a single layout pass.

Reply:
https://bugs.webkit.org/show_bug.cgi?id=249024